JavaScript/JavaScriptSDK/Telemetry/Common/DataSanitizer.ts

```diff
@@ -77,7 +77,7 @@ module Microsoft.ApplicationInsights.Telemetry.Common {
             if (value) {
                 value = Util.trim(value);
                 if (value.toString().length > DataSanitizer.MAX_STRING_LENGTH) {
-                    value = value.substring(0, DataSanitizer.MAX_STRING_LENGTH);
+                    value = value.toString().substring(0, DataSanitizer.MAX_STRING_LENGTH);
```

Comment:
substring function takes start and *not-exclusive* end, so in this case it should actually be substring(0, DataSanitizer.MAX_STRING_LENGTH+1).
Overall I would rather switch it here and in other places to use substr, which behaves as you would expect - takes start_index and length.